### PR TITLE
Grammatical Tweaks (Section 05-06)

### PR DESCRIPTION
**Pull request by furey**
Following https://github.com/adonisjs/docs/pull/314, this PR applies numerous grammatical tweaks to sections 05-06 of the documentation.

### CODE REVIEW

File: 05-Security/02-Authentication.adoc [hunk 1]
```diff
@@ -8,55 +8,66 @@ category: security
 
 toc::[]
 
-Every Web application deals with User Management and Authentication at some stage. The AdonisJs Authentication provider is a fully featured system to authenticate HTTP requests using multiple authenticators. Using authenticators, you can build traditional *session based* login systems to secure *REST API's*.
+The AdonisJs auth provider is a fully featured system to authenticate HTTP requests using multiple authenticators.
+
+Using authenticators you can build traditional *session based* login systems and secure your *REST APIs*.
```
Comment (RomainLanz) [inline 05-Security/02-Authentication.adoc]: We should remove the term `REST` here.
```suggestion

Using authenticators you can build traditional *session based* login systems and secure your *APIs*.
```

File: 05-Security/01-Getting-Started.adoc [hunk 2]
```diff
@@ -8,27 +8,32 @@ category: security
...
-== Forms & Views
-To keep the development cycle simple and productive, AdonisJs ships with some features that you may want to consider before releasing your website to the public.
+[ul-spaced]
+* The `httpOnly` value should be set to `true`, as setting it to `false` will make your cookies accessible via Javascript using `document.cookie`.
+* The `sameSite` value should also be set to `true`, ensuring your session cookie is not visible/accessible via different domains.
 
-=== Form Method Spoofing
-HTML forms are capable of only making *GET* and *POST* requests, which means you cannot make use of all HTTP verbs to perform RESTful operations.
+== Form Method Spoofing
+As HTML forms are only capable of making `GET` and `POST` requests, you cannot use HTTP verbs like `PUT` or `DELETE` to perform RESTful operations via a form's `method` attribute.
```
Comment (RomainLanz) [inline 05-Security/01-Getting-Started.adoc]: Let's remove the `REST` term here.

```suggestion
As HTML forms are only capable of making `GET` and `POST` requests, you cannot use HTTP verbs like `PUT` or `DELETE` to perform resourceful operations via a form's `method` attribute.
```

File: 05-Security/02-Authentication.adoc [hunk 3]
```diff
@@ -8,55 +8,66 @@ category: security
...
+* Lucid (`lucid`)
+* Database (`database`)
+
+== Authentication Categories
+
+Authentication is divided into two broad categories: *Stateful* and *Stateless*.
+
+Session-based authentication is considered *Stateful* since logged in users can navigate to different areas of the application without resending their credentials.
+
+*Stateless* authentication requires the user to resend their credentials on each HTTP request, which is very common with REST APIs.
```
Comment (RomainLanz) [inline 05-Security/02-Authentication.adoc]: Also here 

```suggestion
*Stateless* authentication requires the user to resend their credentials on each HTTP request, which is very common with APIs.
```

File: 05-Security/02-Authentication.adoc [hunk 4]
```diff
@@ -70,63 +81,42 @@ const namedMiddleware = {
...
-Let's start by creating the `login` method on the UserController, which takes the user credentials and log in the user if they are correct.
+Add the `login` method to the `UserController`:
 
+.app/Controllers/Http/UserController.js
 [source, js]
 ----
 class UserController {
 
-  async login ({ request, auth }) {
+  async login ({ request, auth, response }) {
```
Comment (RomainLanz) [inline 05-Security/02-Authentication.adoc]: There's no need to extract the `response` from the context here.
Also, could be good to keep them in the alphabetical order.

```suggestion
  async login ({ auth, request }) {
```

**Review by RomainLanz — CHANGES_REQUESTED** [review]
Wow! Thanks again for this PR @furey! 😍 

I believe we can also add a link to [Adonis Persona](https://github.com/adonisjs/adonis-persona) in the authentication section as a `note` or `tips`.

**Comment by furey** [comment]
> I believe we can also add a link to Adonis Persona in the authentication section as a note or tips.

✅ https://github.com/adonisjs/docs/pull/328/commits/7b6a76bc53250cc1e01081259eeec5d9a5dd611b

**Review by furey — COMMENTED** [review]
> Wow! Thanks again for this PR @furey! 😍

No worries!

**Review by RomainLanz — APPROVED** [review]

**Comment by furey** [comment]
Hi @RomainLanz & @thetutlage,

I'd finished parsing the "06-Digging-Deeper" section of the documentation, so thought I'd update this PR with those files while still waiting for approval/merge.

Cheers!

**Comment by thetutlage** [comment]
@furey Do you want us to wait for the other changes, or should I go ahead and merge it right now and you will do a different PR for rest of the changes?

**Comment by furey** [comment]
@thetutlage Happy for you to merge now and I'll submit a different PR for the rest. Thanks! 👍

**Comment by RomainLanz** [comment]
Thanks a lot!

**Pull request merged**